src/bp/core/services/middleware/event-collector.ts

```diff
@@ -15,7 +15,31 @@ type BatchEvent = sdk.IO.StoredEvent & { retry?: number }
 export const LAST_EVENT_STEP = 'completed'
 
 export const addStepToEvent = (step: string, event: sdk.IO.Event) => {
-  event.processing = { ...(event.processing || {}), [step]: new Date() }
+  if (!event.debugger) {
+    return
+  }
+
+  event.processing = {
+    ...(event.processing || {}),
+    [step]: {
+      ...(event.activeProcessing || {}),
+      date: new Date()
+    }
+  }
+
+  event.activeProcessing = {}
+}
+
+export const addLogToEvent = (logEntry: string, event: sdk.IO.Event) => {
+  if (event.debugger) {
```

Comment:
`event?.debugger && event?.activeProcessing` (let's not assume consumers of the fn provided a correct event)